src/Symfony/Component/ExpressionLanguage/ParserCache/AdapterParserCache.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ExpressionLanguage\ParserCache;
+
+use Symfony\Component\Cache\Adapter\AdapterInterface;
+use Symfony\Component\ExpressionLanguage\ParsedExpression;
+
+/**
+ * @author Alexandre GESLIN <alexandre@gesl.in>
+ */
+class AdapterParserCache implements ParserCacheInterface
+{
+    /**
+     * @var AdapterInterface
+     */
+    private $adapter;
+
+    public function __construct(AdapterInterface $adapter)
+    {
+        $this->adapter = $adapter;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fetch($key)
+    {
+        $item = $this->adapter->getItem($key)
+        if (true !== $item->isHit()) {
```

Comment:
because of the way it's used here, the isHit check could be removed:
`return $this->adapter->getItem($key)->get();`